scripts/js_ts_linter.py

```diff
@@ -0,0 +1,1065 @@
+        if not os.path.exists(eslint_path):
+            python_utils.PRINT('')
+            python_utils.PRINT(
+                'ERROR    Please run start.sh first to install node-eslint ')
+            python_utils.PRINT(
+                '         and its dependencies.')
+            sys.exit(1)
+
+        files_to_lint = self.all_filepaths
+        start_time = time.time()
+        num_files_with_errors = 0
+        summary_messages = []
+
+        num_js_and_ts_files = len(files_to_lint)
+        python_utils.PRINT('Total js and ts files: ', num_js_and_ts_files)
+        eslint_cmd_args = [node_path, eslint_path, '--quiet']
+        result_list = []
+        python_utils.PRINT('Linting JS and TS files.')
+        for _, filepath in enumerate(files_to_lint):
+            if self.verbose_mode_enabled:
+                python_utils.PRINT('Linting: ', filepath)
+            proc_args = eslint_cmd_args + [filepath]
+            proc = subprocess.Popen(
+                proc_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+            encoded_linter_stdout, encoded_linter_stderr = proc.communicate()
+            linter_stdout = encoded_linter_stdout.decode(encoding='utf-8')
+            linter_stderr = encoded_linter_stderr.decode(encoding='utf-8')
+            if linter_stderr:
+                python_utils.PRINT('LINTER FAILED')
+                python_utils.PRINT(linter_stderr)
+                sys.exit(1)
+
+            if linter_stdout:
+                num_files_with_errors += 1
+                result_list.append(linter_stdout)
+
+        if num_files_with_errors:
+            for error in result_list:
+                python_utils.PRINT(error)
+                summary_messages.append(error)
+            summary_message = ('%s    %s JavaScript and Typescript files' % (
+                _MESSAGE_TYPE_FAILED, num_files_with_errors))
+        else:
+            summary_message = (
+                '%s   %s JavaScript and Typescript files linted (%.1f secs)' % (
+                    _MESSAGE_TYPE_SUCCESS, num_js_and_ts_files,
+                    time.time() - start_time))
+        python_utils.PRINT(summary_message)
+        summary_messages.append(summary_message)
+
+        python_utils.PRINT('Js and Ts linting finished.')
+
+        return summary_messages
+
+    def perform_all_lint_checks(self):
+        """Perform all the lint checks and returns the messages returned by all
+        the checks.
+
+        Returns:
+            all_messages: str. All the messages returned by the lint checks.
+        """
+        all_messages = []
+        if not self.all_filepaths:
+            python_utils.PRINT('')
+            python_utils.PRINT(
+                'There are no JavaScript or Typescript files to lint.')
+            return all_messages
+
+        js_and_ts_linter_messages = self._lint_js_and_ts_files()
+        all_messages += js_and_ts_linter_messages
+
+        return all_messages
+
+
+def perform_all_lint_checks(
```

Comment:
Let the main linting function handle this.

Reply:
Removed.